cmd/rqlite/http/env.go

```diff
@@ -0,0 +1,42 @@
+package http
+
+import (
+	"net/url"
+	"os"
+	"strconv"
+)
+
+func ParseHostEnv() (protocol, host string, port uint16, succ bool) {
+	succ = false
+
+	hostEnv := os.Getenv("RQLITE_HOST")
+	uri, err := url.Parse(hostEnv)
+	if err != nil {
+		return
+	}
+
+	if uri.Scheme == "" {
+		protocol = "http"
+	} else if uri.Scheme != "http" && uri.Scheme != "https" {
+		return
+	} else {
+		protocol = uri.Scheme
+	}
+
+	if uri.Hostname() != "" {
```

Comment:
Code would be simpler if it was just `if url.Hostname() == "" return` then you don't need the `else`, right?